modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/IgniteSchema.java

```diff
@@ -41,6 +41,9 @@ public class IgniteSchema extends AbstractSchema {
     /** */
     private final Multimap<String, Function> funcMap = Multimaps.synchronizedMultimap(HashMultimap.create());
 
+    /** */
+    private final Map<String, String> viewMap = new ConcurrentHashMap<>();
```

Comment:
For the tables and the functions we store interfaces like `Function` or `IgniteTable`. For views a string/sql/request instead. Could we similary keep something like `TableMacro`. If I'm right, function `register` would not be required in that case.